### PR TITLE
Scheduler 

### PR DESCRIPTION
**Pull request by teomrd**
Introduces the Schedule page

* Which is manageable via the Netlify CMS
* Displays the Schedule per weekday 
* Includes Broadcast-specific (optional) page

ararres fm

_Styling and other adjustments to follow in secondary PR_

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -0,0 +1,2833 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (stathismor) [inline yarn.lock]: I think you ran `yarn` by mistake and added this lock file.

File: package-lock.json [hunk 2]
```diff
@@ -1,7937 +0,0 @@
-{
```
Comment (stathismor) [inline package-lock.json:1]: Aren't we missing the `package.json`? Or probably not, did you add a new dependency?

File: public/admin/config.yml [hunk 3]
```diff
@@ -7,6 +7,19 @@ media_folder: "public/images/" # Media files will be stored in the repo under st
...
+  - name: "broadcasts"
+    label: "Schedule"
+    folder: "schedule/broadcasts"
+    create: true
+    slug: "{{slug}}"
+    fields:
+      - { label: "Title", name: "title", widget: "string" }
+      - { label: "Subtitle", name: "subtitle", widget: "string" }
+      - { label: "Weekday", name: "weekday", widget: "select", options: ["Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"] }
+      - { label: "Starts", name: "startTime", widget: "string" }
```
Comment (stathismor) [inline public/admin/config.yml:19]: Shouldn't these be `datetime` widgets?

File: pages/programmes/[slug].tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+      <motion.div variants={childVariants}>
+        <article className="prose prose-lg dark:prose-invert max-w-screen-lg m-auto">
+          <ReactMarkdown>{broadcast.content}</ReactMarkdown>
+        </article>
+      </motion.div>
+    </EnterTransition>
+  );
+};
+
+export async function getStaticPaths() {
```
Comment (stathismor) [inline pages/programmes/[slug].tsx:40]: We could probably make this a common function later.

File: lib/broadcasts.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+  const broadcastsPath = fs.readdirSync(path.resolve(BROADCAST_DATA_FOLDER));
+
+  const broadcastsData = broadcastsPath.map((programPath) => {
+    return getMatterData(programPath);
+  });
+
+  return broadcastsData;
+}
+
+export function getBroadcastData(slug: string): BroadcastData {
```
Comment (stathismor) [inline lib/broadcasts.ts:33]: Some of these could be extracted to shared functions later.

**Review by stathismor — COMMENTED** [review]
Yeeey, we have a schedule! :heart:

File: package-lock.json (hunk 2 above)
Comment (teomrd) [reply package-lock.json:1]: Mistakenly switched to `yarn`. It's the muscle memory that did not mind. Reverted it back to `npm`

File: public/admin/config.yml (hunk 3 above)
Comment (teomrd) [reply public/admin/config.yml:19]: Yes I was planning to, or even a pattern widget but couldn't play around with it wittout having that on the admin panel, I am planning to do that on the secondary PR

File: lib/broadcasts.ts (hunk 5 above)
Comment (teomrd) [reply lib/broadcasts.ts:33]: Sure let's get a form and will gather these in a common folder to avoid repeating ourselves :)

**Review by stathismor — APPROVED** [review]
Amazing, feel free to merge!

**Pull request merged**